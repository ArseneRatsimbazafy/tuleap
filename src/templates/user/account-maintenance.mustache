<div id="account-handler" class="row-fluid">
    <div id="account-maintenance" class="span7">
        <ul class="nav nav-tabs">
        {{# tabs }}
            <li {{# is_active }}class="active"{{/ is_active }}><a href="{{ href }}">{{ label }}</a></li>
        {{/ tabs }}
        </ul>

        {{> info }}
        {{> shell }}
        {{> details }}
        {{> third-party-applications }}
        {{{ user_legal }}}
    </div>

    <div id="account-preferences" class="span5">
        {{> preferences }}
    </div>
</div>
