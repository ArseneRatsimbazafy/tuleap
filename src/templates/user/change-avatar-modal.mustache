<div id="change-avatar-modal"
     class="modal hide fade"
     tabindex="-1"
     role="dialog"
     aria-labelledby="change-avatar-modal-title"
     aria-hidden="true"
>
    <div class="modal-header">
        <h3>
            <i class="fa fa-user"></i>
            <span id="change-avatar-modal-title">{{ change_avatar }}</span></h3>
    </div>

    <form action="/account/avatar" method="POST" enctype="multipart/form-data">
        <div class="modal-body">
            <p>{{ change_avatar_desc }}</p>
            <section class="change-avatar-modal-content">
                {{{ avatar }}}
                <div class="change-avatar-modal-actions">
                    <span class="btn change-avatar-modal-actions-select">
                        <i class="fa fa-upload"></i>
                        {{ select_avatar }}
                        <input type="file"
                               name="avatar"
                               id="change-avatar-modal-actions-select-file"
                               class="change-avatar-modal-actions-select-file"
                               accept="image/*">
                    </span>
                    <button type="button" class="btn" id="use-default-avatar-btn">
                        <i class="fa fa-trash-o"></i>
                        {{ use_default_avatar }}
                    </button>
                    <input type="hidden" name="use-default-avatar" value="0" id="use-default-avatar">
                </div>
            </section>
            {{{ csrf_input_html }}}
        </div>

        <div class="modal-footer">
            <button class="btn" data-dismiss="modal" aria-hidden="true">{{ btn_close_label }}</button>
            <button type="submit" class="btn btn-primary" name="add-keys">
                <i class="fa fa-floppy-o"></i>
                {{ btn_save_avatar_label }}
            </button>
        </div>
    </form>
</div>
