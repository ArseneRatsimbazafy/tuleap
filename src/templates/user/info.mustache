<div id="account-user-info">
    <div class="first-user-info">
        <div id="user-avatar">
            {{{ avatar }}}
            <a href="#change-avatar-modal" data-toggle="modal" class="btn btn-small user-avatar-change">
                {{ change_avatar }}
            </a>
        </div>
        <div>
            <h1>
                {{ real_name }}
                {{# can_change_real_name }}
                    <a href="change_realname.php" class="btn btn-small user-real-name-change">
                        {{ change_real_name }}
                    </a>
                {{/ can_change_real_name }}
            </h1>
            <span class="user-username">@{{ user_username }}</span>
        </div>
    </div>

    <table class="user-info-table">
        <tr>
            <td class="user-info-label table-label">{{ user_id_label }}</td>
            <td>{{ user_id_value }}</td>
        </tr>
        <tr>
            <td class="user-info-label table-label">{{ user_email_label }}</td>
            <td>{{ user_email_value }} {{# can_change_email }}<a href="change_email.php" class="btn btn-small">{{ change_email }}</a>{{/ can_change_email }}</td>
        </tr>
        <tr>
            <td class="user-info-label table-label">{{ member_since_label }}</td>
            <td>{{member_since_value }}</td>
        </tr>
        <tr>
            <td class="user-info-label table-label">{{ timezone_label }}</td>
            <td>{{ timezone_value }} <a href="change_timezone.php" class="btn btn-small">{{ change_timezone }}</a></td>
        </tr>
        {{# extra_user_info }}
        <tr>
            <td class="user-info-label table-label">{{ label }}</td>
            <td>{{{ value }}} {{# can_change }}{{{ change }}}{{/ can_change }}</td>
        </tr>
        {{/ extra_user_info }}
    </table>
</div>

{{> change-avatar-modal }}
