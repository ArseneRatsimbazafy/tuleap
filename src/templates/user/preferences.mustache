<h2>{{ preference_title }}</h2>

<form action="updateprefs.php" method="post" class="form-horizontal acco unt-form">
    {{{ csrf_input_html }}}

        <fieldset>
            <h3><i class="fa fa-user"></i> {{ session_label }}</h3>
            <label class="checkbox">
                <input type="checkbox"
                       name="form_sticky_login"
                       value="1"
                       {{# user_has_sticky_login }} checked="checked" {{/ user_has_sticky_login }}
                />
                {{{ remember_me }}}
            </label>
        </fieldset>

        <fieldset>
            <h3><i class="icon-desktop"></i> {{ appearance_title }}</h3>
            <div class="control-group user-account-group">
                <label class="control-label user-account-label">{{# gettext }}Display density:{{/ gettext }}</label>
                <div class="controls">
                    <select name="{{ display_density_name }}" class="select-user-preferences">
                        <option value="">{{# gettext }}Comfortable{{/ gettext }}</option>
                        <option value="{{ display_density_condensed }}" {{# is_condensed }}selected{{/ is_condensed }}>
                            {{# gettext }}Condensed{{/ gettext }}
                        </option>
                    </select>
                </div>
            </div>
            <div class="control-group user-account-group">
                <label class="control-label user-account-label">{{ username_display_label }}</label>
                <div class="controls">
                    <select name="username_display" class="select-user-preferences">
                        {{# user_helper_preferences }}
                            <option value="{{ preference_name }}"
                                    {{# is_selected }} selected="selected" {{/ is_selected }}
                            > {{ preference_label }}
                            </option>
                        {{/ user_helper_preferences }}
                    </select>
                </div>
            </div>

            {{# plugins_prefs }}
                <div class="control-group user-account-group">
                    <label class="control-label user-account-label">{{ name }}</label>
                    <div class="controls">
                        {{{ value }}}
                    </div>
                </div>
            {{/ plugins_prefs }}
        </fieldset>

        <fieldset>
            <h3><i class="fa fa-pencil-square-o"></i> {{ edition_title }} </h3>
            <div class="control-group user-account-group">
                <p>{{ default_format_intro }}</p>
                <label class="control-label user-account-label">{{ default_format_label }}</label>
                <div class="controls">
                    <select name="user_text_default_format" class="select-user-preferences">
                        {{# default_formats }}
                        <option {{# selected }} selected="selected" {{/ selected }}
                            value="{{ value }}"
                        >{{ label }}</option>
                        {{/ default_formats }}
                    </select>
                </div>
            </div>
        </fieldset>

        <fieldset>
            <h3><i class="fa fa-exchange"></i> {{ import_export_title }}</h3>
            <div class="control-group user-account-group">
                <label class="control-label user-account-label">{{{ csv_separator_label }}}</label>
                <div class="controls">
                    <select name="user_csv_separator" class="select-user-preferences">
                        {{# all_csv_separator }}
                            <option value="{{ separator_name }}"
                                    {{# is_selected }} selected="selected" {{/ is_selected }}
                            > {{ separator_label }}
                            </option>
                        {{/ all_csv_separator }}
                    </select>

                    <a href="javascript:help_window('/doc/{{ user_language }}/user-guide/citizen.html#account-maintenance')">
                        <i class="fa fa-question-circle help-button"></i>
                    </a>
                </div>
            </div>

            <div class="control-group user-account-group">
                <label class="control-label user-account-label">{{{ csv_dateformat_label }}}</label>
                <div class="controls">
                    <select name="user_csv_dateformat" class="select-user-preferences">
                        {{# all_csv_dateformat }}
                            <option value="{{ dateformat_name }}"
                                    {{# is_selected }} selected="selected" {{/ is_selected }}
                            > {{ dateformat_label }}
                            </option>
                        {{/ all_csv_dateformat }}
                    </select>

                    <a href="javascript:help_window('/doc/{{ user_language }}/user-guide/citizen.html#account-maintenance')">
                        <i class="fa fa-question-circle help-button"></i>
                    </a>
                </div>
            </div>
        </fieldset>

    <fieldset>
        <h3><i class="fa fa-eye-slash"></i> {{# gettext }}Accessibility{{/ gettext }}</h3>
        <p>
            {{# gettext }}This option displays special patterns on colored cards in Agiledashboard and Cardwall tracker reports. This helps distinguish between colors and their variants.{{/ gettext }}
        </p>
        <label class="checkbox">
            <input type="checkbox"
                   name="form_accessibility_mode"
                   value="1"
                   {{# user_has_accessibility_mode }} checked="checked" {{/ user_has_accessibility_mode }}
            />
            {{# gettext }}Enable accessibility mode{{/ gettext }}
        </label>
    </fieldset>

    <hr/>
    <div class="button-centered">
        <input type="submit"
               name="Submit"
               class="btn btn-primary"
               value="{{ preference_save_button }}"
        >
    </div>
</div>
</form>
