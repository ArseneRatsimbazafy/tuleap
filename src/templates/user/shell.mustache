<div id="account-shell">
    <h3 class="account-title">
        <i class="fa fa-terminal"></i> {{ shell_account_title }}
        <a href="javascript:help_window('/doc/{{ user_language }}/user-guide/citizen.html#account-maintenance')">
            <i class="fa fa-question-circle help-button"></i>
        </a>
    </h3>

    {{{ ssh_keys_extra_html }}}
</div>
