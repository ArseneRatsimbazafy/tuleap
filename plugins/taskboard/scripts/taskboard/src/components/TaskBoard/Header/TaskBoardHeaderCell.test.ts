/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import TaskBoardHeaderCell from "./TaskBoardHeaderCell.vue";
import { createStoreMock } from "@tuleap-vue-components/store-wrapper-jest";
import { UserState } from "../../../store/user/type";

describe("TaskBoardHeaderCell", () => {
    it("displays a cell without color", () => {
        const wrapper = shallowMount(TaskBoardHeaderCell, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_is_admin: false } as UserState } })
            },
            propsData: { column: { id: 2, label: "To do", color: "" } }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays a cell with color", () => {
        const wrapper = shallowMount(TaskBoardHeaderCell, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_is_admin: false } as UserState } })
            },
            propsData: { column: { id: 2, label: "To do", color: "fiesta-red" } }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays a cell with default color", () => {
        const wrapper = shallowMount(TaskBoardHeaderCell, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_is_admin: false } as UserState } })
            },
            propsData: { column: { id: 2, label: "To do", color: "#F8F8F8" } }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays a cell with legacy color to regular users", () => {
        const wrapper = shallowMount(TaskBoardHeaderCell, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_is_admin: false } as UserState } })
            },
            propsData: { column: { id: 2, label: "To do", color: "#87DBEF" } }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
    it("displays a cell with legacy color to admin users", () => {
        const wrapper = shallowMount(TaskBoardHeaderCell, {
            mocks: {
                $store: createStoreMock({ state: { user: { user_is_admin: true } as UserState } })
            },
            propsData: { column: { id: 2, label: "To do", color: "#87DBEF" } }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
