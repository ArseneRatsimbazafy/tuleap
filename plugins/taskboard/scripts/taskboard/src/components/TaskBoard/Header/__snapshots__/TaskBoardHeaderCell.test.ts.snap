// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskBoardHeaderCell displays a cell with color 1`] = `
<div
  class="taskboard-header taskboard-header-fiesta-red"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <span
    class="taskboard-header-label"
  >
    To do
  </span>
   
  <!---->
</div>
`;

exports[`TaskBoardHeaderCell displays a cell with default color 1`] = `
<div
  class="taskboard-header"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <span
    class="taskboard-header-label"
  >
    To do
  </span>
   
  <!---->
</div>
`;

exports[`TaskBoardHeaderCell displays a cell with legacy color to admin users 1`] = `
<div
  class="taskboard-header"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <span
    class="taskboard-header-label"
  >
    To do
  </span>
   
  <wrong-color-popover-stub
    color="#87DBEF"
  />
</div>
`;

exports[`TaskBoardHeaderCell displays a cell with legacy color to regular users 1`] = `
<div
  class="taskboard-header"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <span
    class="taskboard-header-label"
  >
    To do
  </span>
   
  <!---->
</div>
`;

exports[`TaskBoardHeaderCell displays a cell without color 1`] = `
<div
  class="taskboard-header"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <span
    class="taskboard-header-label"
  >
    To do
  </span>
   
  <!---->
</div>
`;

exports[`TaskBoardHeaderCell does not display wrong color popover nor label if column is collapsed 1`] = `
<div
  class="taskboard-header taskboard-header-collapsed"
>
  <expand-button-stub
    column="[object Object]"
  />
   
  <collapse-button-stub
    column="[object Object]"
  />
   
  <!---->
   
  <!---->
</div>
`;
